scanner/src/main/kotlin/scanners/ScanCode.kt

```diff
@@ -388,71 +390,101 @@ class ScanCode(name: String, config: ScannerConfiguration) : LocalScanner(name,
+                        endLine = it["end_line"].intValue()
+                    )
+                )
+            }
+        }
+
+    /**
+     * Get the copyright findings from the given [result].
+     */
+    internal fun getCopyrightFindings(result: JsonNode): List<CopyrightFinding> =
+        result["files"].flatMap { file ->
+            val path = file["path"].textValue()
+
+            file["copyrights"]?.toList().orEmpty().flatMap { copyrights ->
+                val startLine = copyrights["start_line"].intValue()
+                val endLine = copyrights["end_line"].intValue()
+                // While ScanCode 2.9.2 was still using "statements", version 2.9.7 is using "value".
+                val statements = (copyrights["statements"] ?: listOf(copyrights["value"]))
+
+                statements.map { statement ->
+                    CopyrightFinding(
+                        statement = statement.textValue(),
+                        location = TextLocation(
+                            path = path,
+                            startLine = startLine,
+                            endLine = endLine
+                        )
+                    )
+                }
+            }
+        }
+
     /**
      * Get the license found in one of the commonly named license files, if any, or an empty string otherwise.
      */
-    internal fun getRootLicense(result: JsonNode): String {
+    internal fun getRootLicense(licenseFindings: List<LicenseFinding>): String =
         // TODO: This function should return a list of all licenses found in all license files instead of only a single
         //       license.
-
-        val rootLicenseFile = result["files"].singleOrNull { file ->
-            val path = file["path"].textValue()
+        licenseFindings.singleOrNull { finding ->
             try {
-                file["licenses"]?.singleOrNull() != null
-                        && LICENSE_FILE_MATCHERS.any { it.matches(Paths.get(path)) }
+                LICENSE_FILE_MATCHERS.any { it.matches(Paths.get(finding.location.path)) }
             } catch (e: InvalidPathException) {
                 false
             }
-        } ?: return ""
-
-        return getLicenseId(rootLicenseFile["licenses"].single())
-    }
+        }?.license ?: ""
 
     /**
      * Return the copyright statements in the vicinity, as specified by [toleranceLines], of [licenseStartLine] in the
      * file [path]. The default value of [toleranceLines] is set to 5 which seems to be a good balance between
      * associating findings separated by blank lines but not skipping complete license statements.
      */
     internal fun getClosestCopyrightStatements(
-        path: String,
-        copyrights: Collection<JsonNode>,
+        copyrights: List<CopyrightFinding>,
         licenseStartLine: Int,
         toleranceLines: Int = 5
     ): SortedSet<CopyrightFindings> {
+        require(copyrights.map { it.location.path }.distinct().size <= 1)
```

Comment:
This will throw an `IllegalArgumentException` if the condition is false. Is this intended? If so, please add a meaningful message instead of the default "Failed requirement".

Reply:
This is intended, yes. Isn't the `intention` quite obvious?
The exception log will point you to the line of code from which you can imo extract why this is thrown. It requires that the given findings are for a single file.

I prefer not doing that as I find it redundant.

Reply:
I mean: `copyrights.map { it.location.path }.distinct().size` equals number of distinct files.
Imo the code is quite readable.

Reply:
I was wondering if it's intentional because you didn't provide a message. Usually you want users to know what's wrong without having to look at the source code. But I understand this is just an intermediate step in the refactoring.

Reply:
I think this is a typical pre-condition assertion which should never throw. In case it throws it indicates a `bug`. And the target `audience` of bugs I guess are developers.